src/commands/music/index.js

```diff
@@ -0,0 +1,215 @@
+import Promise from 'bluebird';
+import R from 'ramda';
+import ytdl from 'ytdl-core';
+
+import { addSong, delSong, delSongs, getSong, getSongs, getNextSong } from '../../redis';
+import { time } from '../../helpers';
+import T from '../../translate';
+
+let timer;
+
+
+function vleave(client, evt, lang) {
+  const info = client.VoiceConnections.getForGuild(evt.message.guild);
+  if (info) {
+    info.voiceConnection.disconnect();
+    delSongs(evt.message.guild.id);
+    delSong(evt.message.guild.id);
+  } else {
+    return Promise.resolve(T('music_not_connected', lang));
+  }
+}
+
+function vjoin(client, evt, suffix, lang) {
+  if (!suffix) {
+    const channel = evt.message.guild.voiceChannels.filter(vc => vc.members.map(m => m.id).indexOf(evt.message.author.id) > -1);
+    if (channel.length > 0) {
+      channel[0].join();
+      timer = setTimeout(vleave, 1200000, client, evt);
```

Comment:
Maybe move this in to a function instead so it's easier to modify the leave timeout.